### DISCUSSION TITLE
Allow `ignorePresets` to be propagated throuh nested presets

### DISCUSSION THREAD

**Discussion by secustor** [renovatebot/renovate#31058] [Suggest an Idea]
### Tell us more.

Currently `ignorePresets` has to be in the same config as the `extends` which introduces the preset which should be ignored. 

In case of centrally managed presets that can become an issue. 
e.g. a the repo config extends `config:best-practices` but at the same time via organizational preset e.g. a replacement should be ignored.

**Comment by rarkins** [comment]
Can you give more details of an example setup?

**Reply by secustor** [reply]
[Repo A consume a shared preset](https://github.com/secustor/backstage-plugins/blob/main/renovate.json5#L6) which sets up some common settings for Backstage. 
Because of limitations of the upstream project it does not make sense to create replacements PRs for MaterialUI. 
Therefore I would like to set the preset up that everyone who uses it ignores that specific preset. 
https://github.com/secustor/backstage-renovate-config/blob/c4b53d0b9b990ed957baa70bfe239a36ec44229f/app.json5#L7-L9

Though `ignorePreset` works only for extens in the same file as they are removed during config migrations.

**Reply by rarkins** [reply]
I had thought that ignorePresets should work for nested presets already. I even wonder if it once did, because ignoring nested presets was one of the main intentions of it. You're welcome to fix it of course, alternatively if you can have a reproduction demonstrating it then I can try debugging

**Reply by secustor** [reply]
This are two different scenarios. The one you are describing the `ignorePreset` is in the repo config and the `extend` in the preset. I want to allow the other way around too.

**Reply by rarkins** [reply]
That's a tough one because the repo config is resolved first

**Reply by secustor** [reply]
Never said it is easy 😅    
I simply wanted to discuss if this is something we should support.

**Reply by rarkins** [reply]
I think not - we need to process in the order that presets are resolved

**Reply by Finkregh** [reply]
I just ran into the same issue, my employer is providing a central renovate which uses `config:recommended` which includes `:ignoreModulesAndTests`, which I'd like to ignore as i.e. `/tests/requirements.txt` should be still be updated.

We (org level) mostly have centrally managed renovate configs ourselves, which I guessed would allow `ignorePresets`, but it seems that only works when it is applied in each repo's own config.

Is there any way to override the `ignorePaths` via a central config which is loaded via `extends`?

**Reply by Finkregh** [reply]
using `ignorePaths` directly instead of `ignorePresets` behaves the same; works in repo's own config, not in one loaded via extends :(
